Mathlib/Data/Finset/Sum.lean

```diff
@@ -111,8 +110,7 @@ forms a quasi-inverse to `disjSum`, in that it recovers its left input.
 See also `List.partitionMap`.
 -/
 def toLeft (s : Finset (α ⊕ β)) : Finset α :=
-  s.disjiUnion (Sum.elim singleton (fun _ => ∅)) <| by
-    simp [Set.PairwiseDisjoint, Set.Pairwise, Function.onFun, eq_comm]
+  s.filterMap (Sum.elim some fun _ => none) (by clear x; aesop)
```

Comment:
Is there a tracking issue for `aesop` pulling in `x` here?

If `aesop` changes (fixes) its behaviour, this might error, right?

Reply:
A fix to aesop would make this clear unnecessary, but it wouldn't make it an error